src/writer.rs

```diff
@@ -169,19 +170,41 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
-    /// Write event nested in another level:
-    /// ```xml
-    /// <parent_name>{event}</parent_name>
-    /// ```
-    pub fn write_nested_event<'a, N: AsRef<[u8]>, E: AsRef<Event<'a>>>(
-        &mut self,
-        parent_name: N,
-        event: E,
-    ) -> Result<()> {
-        let name = parent_name.as_ref();
-        self.write_event(Event::Start(BytesStart::borrowed(name, name.len())))?;
-        self.write_event(event)?;
-        self.write_event(Event::End(BytesEnd::borrowed(name)))?;
+    /// Write nested
+    pub fn create_node<'a, N: 'a + AsRef<[u8]>>(&'a mut self, name: &'a N) -> NodeWriter<W> {
+        NodeWriter {
+            writer: self,
+            tag_name: name.as_ref(),
+            start_tag: BytesStart::borrowed_name(name.as_ref()),
+        }
+    }
+}
+
+pub struct NodeWriter<'a, W: Write> {
+    writer: &'a mut Writer<W>,
+    tag_name: &'a [u8],
```

Comment:
As I mentioned, I'm kind of a novice with lifetimes, but I couldn't figure out how to re-use the value from ```start_tag``` without compile errors.